app/controllers/api/v1/api_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class ApiController < ApplicationController
+      include Concerns::ExceptionHandler
+
+      # Handle AuthenticationErrors
+      rescue_from RailsInterviewProject::Errors::UnauthorizedRequestError, with: :handle_unauthorized_error
+
+      private
+
+      # Internal: Mock API authentication for controller actions
+      #
+      # Raises TenantNotFound exception on invalid token lookup
```

Comment:
`TenantNotFound ` to `UnauthorizedRequestError`.

Reply:
https://github.com/SpicyCookies/rails-interview-project/pull/5/commits/2074912134a80bb7cc37069e68ee80d3f2cc2d3c